### ISSUE TITLE
logging data races

### ISSUE THREAD

**Issue by jrick** [planetdecred/godcr#455]
Enabling the race detector shows there are a lot of races happening around the logging, and this is the cause:

```
// Write writes the data in p to standard out and the log rotator.
func (l logWriter) Write(p []byte) (n int, err error) {
        os.Stdout.Write(p)
        go func() {
                internalLog <- string(p)
        }()
        return logRotator.Write(p)
}
```

The byte slice passed to Write is being modified later on, because the slog package will recycle and reuse the byte slice to minimize allocations.  Asynchronously reading this slice in another goroutine spawned by Write is incorrect and unsafe.

**Comment by oluwandabira** [comment]
Is the method of resolving this (copying the bytes) proposed in pr #525 acceptable?

**Comment by Sirmorrison** [comment]
Resolved by PR #574
